data_utils/host.py

```diff
@@ -18,37 +18,207 @@
+        owner: A string containing the repository owner.
+
+    Returns:
+        A JSON object with the pull request reviewer information.
+    """
+
+    query = f"""{{
+        repository(name: "{name}", owner: "{owner}") {{
+            pullRequests(first: 5) {{
+                nodes {{
+                    resourcePath
+                    author {{
+                        login
+                    }}
+                    timelineItems(first: 100) {{
+                        nodes {{
+                            ... on ReviewRequestedEvent {{
+                                requestedReviewer {{
+                                    ... on User {{
+                                        login
+                                    }}
+                                }}
+                            }}
+                            ... on PullRequestReview {{
+                                author {{
+                                    login
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}
+        }}
+    }}"""
+
+    return run_query(query)
+
+
+def process_reviewer_query_results(result, host_dict, intern_usernames):
+    """ Processes the query results for the reviewer usernames.
+
+    Updates host_dict if new host usernames are found. Assumes that reviewers
+    in starter projects are hosts.
+
+    Args:
+        result: The results from the query.
+        host_dict: Dictionary containing host information.
+        intern_usernames: Set of known intern usernames.
+
+    Returns:
+        None.
+    """
+
+    try:
+        pull_requests = (result["data"]["repository"]["pullRequests"]["nodes"])
+    except:
+        raise Exception(
+            "Query results for reviewers does not have a structure that is"
+            " currently supported by RISR.")
+
+    for pull_request in pull_requests:
+        timeline_items = pull_request["timelineItems"]["nodes"]
+        for review_item in timeline_items:
+            if not review_item:
+                continue
+            if "requestedReviewer" in review_item:
+                host = review_item["requestedReviewer"]
+            if "author" in review_item:
+                host = review_item["author"]
+
+            # Special case in which host account has been deleted
+            try:
+                host_username = host["login"]
+            except TypeError:
+                continue
+
+            # Check if host is already in dictionary and if intern reviewed
+            # their own pull request.
+            if host_username not in host_dict and \
+                host_username not in intern_usernames:
+                # Start date and team number are unknown in this case.
+                host_dict[host_username] = ["unknown", "unknown"]
+
+
+def get_hosts_from_teams_csv(teams_file, host_dict):
+    """ Gets host information from the internal STEP teams CSV file.
```

Comment:
Drop "internal" from this comment -- generally don't reference non-public data externally.

Reply:
Done!